pandas/tests/series/test_dtypes.py

```diff
@@ -487,3 +488,241 @@ def test_reindex_astype_order_consistency(self):
+                        (True, False),
+                        (True, False),
+                    ): pd.CategoricalDtype(),
+                },
+            ),
+            (
+                pd.to_datetime(["2020-01-14 10:00", "2020-01-15 11:11"]),
+                pd.DatetimeTZDtype(tz="UTC"),
+                {
+                    (
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                    ): pd.DatetimeTZDtype(tz="UTC"),
+                },
+            ),
+            (
+                pd.to_datetime(["2020-01-14 10:00", "2020-01-15 11:11"]),
+                "datetime64[ns]",
+                {
+                    (
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                    ): np.dtype("datetime64[ns]"),
+                },
+            ),
+            (
+                pd.to_datetime(["2020-01-14 10:00", "2020-01-15 11:11"]),
+                object,
+                {
+                    ((True,), (True, False), (True, False), (True, False),): np.dtype(
+                        "datetime64[ns]"
+                    ),
+                    ((False,), (True, False), (True, False), (True, False),): np.dtype(
+                        "O"
+                    ),
+                },
+            ),
+            (
+                pd.period_range("1/1/2011", freq="M", periods=3),
+                None,
+                {
+                    (
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                    ): pd.PeriodDtype("M"),
+                },
+            ),
+            (
+                pd.arrays.IntervalArray([pd.Interval(0, 1), pd.Interval(1, 5)]),
+                None,
+                {
+                    (
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                        (True, False),
+                    ): pd.IntervalDtype("int64"),
+                },
+            ),
+        ],
+    )
+    @pytest.mark.parametrize("params", product(*[(True, False)] * 4))
+    def test_convert_dtypes(self, data, maindtype, params, answerdict):
+        if maindtype is not None:
+            series = pd.Series(data, dtype=maindtype)
+        else:
+            series = pd.Series(data)
+        answers = {k: a for (kk, a) in answerdict.items() for k in product(*kk)}
+
+        ns = series.convert_dtypes(*params)
+        expected_dtype = answers[tuple(params)]
+        expected = pd.Series(series.values, dtype=expected_dtype)
+        tm.assert_series_equal(ns, expected)
+
+        # Test that it is a copy
+        copy = series.copy(deep=True)
+        ns[ns.notna()] = np.nan
```

Comment:
what exactly is this doing?

Reply:
I am following a pattern in `test_api.py:test_copy`.  I want to make sure that the values backing the original series are not modified when returning the result (i.e., testing that I always return a copy)

Reply:
right this doesn't appear to do anything (ns is not used here)

Reply:
But it is testing whether modifying `ns`, which is the result of `convert_dtypes` is also modifying the original series.

Reply:
ahh ok